### PR TITLE
Add a fortran ring compile and run test to mpi_ring_c_test.sh

### PR DESCRIPTION
**Pull request by wckzhang**
Signed-off-by: William Zhang <wilzhang@amazon.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wckzhang** [comment]
The test works in my personal pipeline, catches the failures and passes normally as expected.

**Review by shijin-aws — APPROVED** [review]

File: mpi_ring_c_test.sh [hunk 1]
```diff
@@ -26,21 +28,32 @@ fi
 
 host_setup ${hostfile} ${hosts}
 mpicc -o /tmp/ring_c ring_c.c
+# Fortran compile test
+mpif90 -o /tmp/ring_usempi ring_usempi.f90
```
Comment (wzamazon) [inline mpi_ring_c_test.sh:32]: Would it make sense to have a separate file ```mpi_ring_fortran_test.sh```?

Comment (wckzhang) [reply mpi_ring_c_test.sh:32]: I think so, I can update it

Comment (wckzhang) [reply mpi_ring_c_test.sh:32]: I'll probably hold off on this though for now since it's not an important change, I'll follow up later on this

Comment (wzamazon) [reply mpi_ring_c_test.sh:32]: Thanks, would you please set the name of the binary to ```ring_fortran```. The current name ```ring_usempi``` is a little confusing.

Comment (wckzhang) [reply mpi_ring_c_test.sh:32]: Sure

File: common.sh [hunk 2]
```diff
@@ -437,6 +437,7 @@ EOF
 alinux_install_deps() {
     cat <<-"EOF" >> ${tmp_script}
     sudo yum -y groupinstall 'Development Tools'
+    sudo yum install -y gcc-gfortran
```
Comment (shijin-aws) [inline common.sh:440]: I think `Development Tools` include gcc and gfortran
```
[ec2-user@ip-172-31-11-68 SubspaceBenchmarks-ompi]$ yum groupinfo "Development Tools"
Loaded plugins: dkms-build-requires, priorities, update-motd, upgrade-helper
1067 packages excluded due to repository priority protections

Group: Development tools
 Group-Id: development
 Description: A basic development environment.
 Mandatory Packages:
   =autoconf
   =automake
    binutils
   =bison
   =flex
   =gcc
   =gcc-c++
   =gdb
   =gettext
   =kexec-tools
   =latrace
   =libtool
    make
   =patch
    pkgconfig
   =rpm-build
   =strace
   =system-rpm-config
   =systemtap-runtime
 Default Packages:
   =byacc
   =crash
   =cscope
   =ctags
   =cvs
   =diffstat
   =doxygen
   =elfutils
   =gcc-gfortran
   =git
   =indent
   =intltool
   =ltrace
   =patchutils
   =rcs
   =subversion
   =swig
   =systemtap
   =texinfo
   =valgrind
 Optional Packages:
   ant
   bzr-python27
   ccache
   chrpath
   clips
   clips-devel
   clips-doc
   clips-emacs
   clips-xclips
   clipsmm-devel
   clipsmm-doc
   cmake
   cmucl
   colordiff
   cvsps
   darcs
   dejagnu
   email2trac
   expect
   ftnchek
   gcc-gnat
   ghc
   git
   glibc-utils
   haskell-platform
   imake
   jpackage-utils
   lslk
   lua
   mercurial
   mercurial-python27
   mock
   mod_dav_svn
   nasm
   nqc
   nqc-doc
   ocaml
   perltidy
   python27-babel
   python27-docs
   qgit
   rpmdevtools
   rpmlint
   sbcl
   systemtap-initscript
   trac
   trac-git-plugin
   trac-mercurial-plugin
   trac-webadmin
   translate-toolkit
```

Comment (wckzhang) [reply common.sh:440]: Yeah they do, (I knew that) I just wanted to be consistent. I can remove it.

**Review by shijin-aws — APPROVED** [review]

**Pull request merged**